staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/builder.go

```diff
@@ -103,14 +103,14 @@ func BuildSwagger(crd *apiextensionsv1.CustomResourceDefinition, version string,
 				if opts.AllowNonStructural || len(structuralschema.ValidateStructural(nil, ss)) == 0 {
 					schema = ss
 
+					schema = schema.Unfold()
```

Comment:
Maybe add a comment that this will add anyOf that needs to be stripped by the line below.

Reply:
Why was this moved? In order to also skip the value validations coming from unfolding?

Reply:
Yes. Unfolding adds the `anyOf` tag if x-int-or-string is set. The tag isn't compatible with the v2 schema so we need to strip them via `schema.StripValueValidations()`

Reply:
lgtm